server/src/main/java/org/candlepin/model/OwnerEnvContentAccess.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.model;
+
+import org.hibernate.annotations.ForeignKey;
+import org.hibernate.annotations.GenericGenerator;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import javax.xml.bind.annotation.XmlRootElement;
+
+
+
+/**
+ * Represents the join table between Environment and Owner.
+ *
+ * This class uses composite primary key from the two
+ * entities. This strategy has been chosen so that
+ * the current Candlepin schema doesn't change. However,
+ * should we encounter any problems with this design,
+ * there is nothing that stops us from using standard
+ * uuid for the link.
+ */
+@XmlRootElement
+@Entity
+@Table(name = "cp_owner_env_content_access")
+public class OwnerEnvContentAccess extends AbstractHibernateObject {
+
+    private static final long serialVersionUID = 4551448607292211792L;
+
+    @Id
+    @GeneratedValue(generator = "system-uuid")
+    @GenericGenerator(name = "system-uuid", strategy = "uuid")
+    @Column(length = 32)
+    @NotNull
+    private String id;
+
+    @ManyToOne
+    @ForeignKey(name = "fk_owner_env_cont_acc_owner")
```

Comment:
Pretty sure [`@ForeignKey`](https://docs.jboss.org/hibernate/jpa/2.1/api/javax/persistence/ForeignKey.html) is only used for schema generation which we don't do.